crypto/src/lib.rs

```diff
@@ -16,6 +16,22 @@
 //!   [RandomCoin] implementation uses a cryptographic hash function to generate pseudo-random
 //!   elements form a seed.
 
+#![no_std]
+
+#[cfg(not(any(feature = "std", feature = "alloc")))]
+compile_error!("Either feature \"std\" or \"alloc\" must be enabled for this crate.");
+
+#[cfg(all(feature = "alloc", feature = "concurrent"))]
+compile_error!("This crate does not support features \"alloc\" and \"concurrent\" simultaneously.");
```

Comment:
Since rayon is not `no_std` compatible, wouldn't it be simpler to make the `concurrent` feature imply `std` and then branch only on `std` vs. `no_std + alloc`? The less conditions the better.

Reply:
That's a possibility indeed. One thing that may need clarity, either in READMEs or through a conditional compiler error is if both alloc and std are allowed simultaneously. We currently require at least 1 of them, but having both will break compilation of the winter-utils crate, so we may need to have this second conditional attribute anyway (changed to `#cfg[not(all(feature = "alloc", feature = "std"))]`)